src/elements/common/annotator-context/withAnnotations.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import AnnotatorContext from './AnnotatorContext';
+import { Action, AnnotationActionEvent, AnnotatorState, Status } from './types';
+
+type WrapperProps = {
+    showAnnotationControls: boolean;
+};
+
+type WrapperReturn<P> = React.ComponentClass<P & WrapperProps>;
+
+export default function withAnnotations<P extends object>(
+    WrappedComponent: React.ComponentType<P & WrapperProps>,
+): WrapperReturn<P> {
+    class ComponentWithAnnotations extends React.Component<P & WrapperProps, AnnotatorState> {
+        static displayName: string;
+
+        state: AnnotatorState = {
+            annotation: null,
+            action: null,
+            error: null,
+        };
+
+        getActionSuffix(status: Status) {
+            return status === Status.SUCCESS || status === Status.ERROR ? 'end' : 'start';
+        }
+
+        handleAnnotationCreate = ({ annotation, error, meta }: AnnotationActionEvent): void => {
```

Comment:
Is the suffix needed or can we infer an error if the `error` property is defined?